ideascube/mediacenter/management/commands/import_medias.py

```diff
@@ -87,23 +89,36 @@ def add(self, metadata):
 
         instance = Document.objects.filter(title=title, kind=kind).last()
         if instance and not self.update:
-            return self.report.warning('Document exists (Use --update for '
-                                       'reimport)', title)
-
-        path = os.path.join(self.ROOT, original)
-        if not os.path.exists(path):
-            return self.report.error(u'Path not found', path)
-        with open(path, 'rb') as f:
-            original = File(f, name=original)
-
+            self.report.warning('Document exists (Use --update for reimport)',
+                                title)
+            return
+
+        original_path = os.path.join(self.ROOT, original)
+        try:
+            with open(original_path, 'rb') as f:
+                original = File(f, name=original)
+
+                preview = metadata.get('preview')
+                if preview:
+                    preview_path = os.path.join(self.ROOT, preview)
+                    try:
+                        with open(preview_path, 'rb') as f:
+                            preview = File(f, name=preview)
+                            self.save(metadata, original, instance, preview)
+                    except IOError:
+                        self.report.error("Cannot open path", preview_path)
+                else:
+                    self.save(metadata, original, instance)
+        except IOError:
+            self.report.error("Cannot open path", original_path)
             preview = metadata.get('preview')
             if preview:
-                path = os.path.join(self.ROOT, preview)
-                with open(path, 'rb') as f:
-                    preview = File(f, name=preview)
-                    self.save(metadata, original, instance, preview)
-            else:
-                self.save(metadata, original, instance)
+                preview_path = os.path.join(self.ROOT, preview)
+                try:
+                    with open(preview_path, 'rb') as f:
+                        pass
```

Comment:
So, since you're not going to import anything here anyway, you don't need to open the preview path at all.

It's certainly nice to let the user know that the preview path doesn't exist here (otherwise they'd fix the original paths then try again and we'd tell them this time the preview paths are missing).

So how about just `if not os.path.isfile(preview_path)`?